community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/BatchingMultipleIndexPopulator.java

```diff
@@ -252,6 +253,15 @@ private void flush( IndexDescriptor descriptor, List<NodePropertyUpdate> batch )
         } );
     }
 
+    private void processPopulationFailuresIfAny()
+    {
+        PopulationFailure item;
+        while ( (item = populationFailures.poll()) != null )
```

Comment:
We need to process all population failures from the queue so each of them should be removed via `#poll()`. Explicit `!isEmpty()` check is not present because `LinkedBlockingQueue#poll()` has it inside as a first check.